xl/xenops_server_xenlight.ml

```diff
@@ -2995,9 +3021,11 @@ let process_one_watch xc xs (path, token) =
 	if path = _introduceDomain || path = _releaseDomain
 	then look_for_different_domains xc xs
 	else match List.filter (fun x -> x <> "") (Xstringext.String.split '/' path) with
-		| "local" :: "domain" :: domid :: "backend" :: kind :: frontend :: devid :: _ ->
+		| "local" :: "domain" :: domid :: "backend" :: kind :: frontend :: devid :: key ->
 			debug "Watch on backend domid: %s kind: %s -> frontend domid: %s devid: %s" domid kind frontend devid;
-			fire_event_on_device frontend kind devid
+			fire_event_on_device frontend kind devid;
+			(* If this event was a state change then this might be the first time we see evidence of PV drivers *)
+			if key = ["state"] then maybe_update_pv_drivers_detected ~xc ~xs (int_of_string frontend) path
```

Comment:
Okay, so you're talking about some further evidence by which PV drivers hint at their existence. If you can find the old check, that would be interesting, and perhaps it could be resurrected in future to bolster the PV driver detection.

However, for now, I'm unconcerned about reporting PV-drivers-detected=false for a VM with no VBDs or VIFs.